core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -1716,7 +1717,29 @@ class WorkspaceService(protected val userInfo: UserInfo, val dataSource: SlickDa
     // Next call GenomicsService, which actually makes the Google Genomics API call.
     .flatMap { _ =>
       val genomicsServiceRef = genomicsServiceConstructor(userInfo)
-      genomicsServiceRef.GetOperation(jobId)
+      genomicsServiceRef.GetOperation("operations/" + jobId)
+    }
+  }
+
+  def getGenomicsOperationV2(workflowId: String, operationId: List[String]): Future[PerRequestMessage] = {
+    // stuff to parse the call level metadata from cromwell so we can gather all the operation ids
+    // for the workflow so we can check that the requested operation id actually exists in the workflow
+    case class Call(jobId: Option[String])
+    case class OpMetadata(calls: Option[Map[String, Seq[Call]]])
+    implicit val callFormat = jsonFormat1(Call)
+    implicit val opMetadataFormat = jsonFormat1(OpMetadata)
+
+    // note that cromiam should only give back metadata if the user is authorized to see it
+    cromiamDAO.callLevelMetadata(workflowId, MetadataParams(includeKeys = Set("jobId")), userInfo).flatMap { metadataJson =>
+      val operationIds = metadataJson.convertTo[OpMetadata].calls.getOrElse(Map.empty).values.flatMap(_.flatMap(_.jobId))
```

Comment:
i think this would benefit from some intermediate variable assignments

Reply:
I tried it and and didn't like it as much, what do you think?

```
    val operationIds = for {
      calls <- json.convertTo[OpMetadata].calls.values
      call <- calls
      jobId <- call.jobId
    } yield jobId
```